WalletWasabi/Crypto/GroupElementVector.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto
+{
+	public class GroupElementVector : IEnumerable<GroupElement>
```

Comment:
In c# classes are always in singular, that's why we see `UserCollection` instead of `Users`. The `Vector` here mean the same that in rust `Vec<GroupElement>`. This is not a Set, Collection nor List. I will think in a better name, we can rename it anytime.

Reply:
Hmm, why isn't it a list, because it's not resizable? And yeah C++ uses the same notion of "vector". Anyway, it's not a big deal IMO

Reply:
just to troll you Yuval! a vector/list of group elements DO form a vector space over the field, they're defined :)

Reply:
sure but then we'll need to change the `+` overload on `GroupElement` to add points by adding their coordinates ;-)

seriously though:

1. `List` in C# is like C++'s `vec` - i.e. variable length, homogenous type
2. "tuple" in C# means fixed length, mixed types
3. "array" means fixed length, homogenous type
4. ... "sequence"? implies something else to me

even though "vector" in C++ is a historical mistake and nothing more, i don't think it's that bad since it's convention by now